File: backend/src/main/java/com/tradewise/auth/service/AuthService.java
package com.tradewise.auth.service;

import com.tradewise.auth.dto.request.ForgotPasswordRequest;
import com.tradewise.auth.dto.request.LoginRequest;
import com.tradewise.auth.dto.request.LogoutRequest;
import com.tradewise.auth.dto.request.RefreshTokenRequest;
import com.tradewise.auth.dto.request.RegisterRequest;
import com.tradewise.auth.dto.request.ResetPasswordRequest;
import com.tradewise.auth.dto.response.AuthResponse;
import com.tradewise.auth.dto.response.OtpResponse;
import com.tradewise.auth.dto.response.TokenResponse;
import com.tradewise.auth.dto.response.UserResponse;

/**
 * Authentication use cases. Defined as an interface to keep controllers
 * decoupled from the implementation (SOLID / DIP).
 */
public interface AuthService {

    UserResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    TokenResponse refreshToken(RefreshTokenRequest request);

    void logout(LogoutRequest request);

    OtpResponse forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);
}

File: frontend/src/features/auth/services/auth.service.ts
import type { ApiResponse } from '../../../types/api'
import { apiClient } from '../../../services/api-client'
import type { AuthResponse, ChangePasswordRequest, LoginRequest, OtpResponse, RegisterRequest, ResetPasswordRequest, TokenResponse, UpdateProfileRequest, User } from '../types/auth.types'

export const authService = {
  login: async (request: LoginRequest) => (await apiClient.post<ApiResponse<AuthResponse>>('/auth/login', request)).data.data,
  register: async (request: RegisterRequest) => (await apiClient.post<ApiResponse<User>>('/auth/register', request)).data.data,
  refresh: async (refreshToken: string) => (await apiClient.post<ApiResponse<TokenResponse>>('/auth/refresh', { refreshToken })).data.data,
  logout: async (refreshToken: string) => { await apiClient.post('/auth/logout', { refreshToken }) },
  forgotPassword: async (email: string) => (await apiClient.post<ApiResponse<OtpResponse>>('/auth/forgot-password', { email })).data.data,
  resetPassword: async (request: ResetPasswordRequest) => { await apiClient.post('/auth/reset-password', request) },
  me: async () => (await apiClient.get<ApiResponse<User>>('/users/me')).data.data,
  updateProfile: async (request: UpdateProfileRequest) => (await apiClient.put<ApiResponse<User>>('/users/me', request)).data.data,
  changePassword: async (request: ChangePasswordRequest) => { await apiClient.put('/users/me/password', request) },
}